index.js

```diff
@@ -23,15 +20,38 @@ module.exports = function(db) {
 
   feed.change = 0
   feed.notify = []
+  feed.batch = []
 
   feed.append = function(value, cb) {
     if (!cb) cb = noop
     if (!Buffer.isBuffer(value)) value = new Buffer(value)
 
-    lock(function(release) {
-      ensureCount(function(err) {
-        if (err) return release(cb, err)
-        db.put(lexint.pack(++feed.change, 'hex'), value, function(err) {
+    ensureCount(function(err) {
+      if (err) return release(cb, err)
+
+      var batch = feed.batch
+      var change = ++feed.change
+      batch.push({
+        change: change,
+        key: lexint.pack(change, 'hex'),
+        value: value,
+        callback: cb
+      })
+
+      if (batch.length !== 1) return
+
+      // schedule batch commit
+      process.nextTick(function () {
+        batch = batch.slice()
+        feed.batch.length = 0
+
+        db.batch(batch.map(function (item) {
+          return {
+            type: 'put',
+            key: item.key,
+            value: item.value
+          }
+        }), function(err) {
```

Comment:
right, the tmp batch we already have, it's `feed.batch`. We just need a lock around the commit so there is never the possibility of commit 2 succeeding while commit1 fails.

the question is what to do when the current batch fails, and there's also a tmp batch in the balance